math/polynomial/polynomial.go

```diff
@@ -0,0 +1,123 @@
+// Package polynomial provides representations of polynomials over the scalars
+// of a group.
+package polynomial
+
+import (
+	"fmt"
+
+	"github.com/cloudflare/circl/group"
+)
+
+// Polynomial stores a polynomial over the set of scalars of a group.
+type Polynomial struct {
+	// Internal representation is in polynomial basis:
+	// Thus,
+	//     p(x) = \sum_i^k c[i] x^i,
+	// where k = len(c)-1 is the degree of the polynomial.
+	c []group.Scalar
+}
+
+// New creates a new polynomial given its coefficients in ascending order.
+// Thus,
+//     p(x) = \sum_i^k c[i] x^i,
+// where k = len(c)-1 is the degree of the polynomial.
+func New(coeffs []group.Scalar) Polynomial {
+	if len(coeffs) == 0 {
+		panic("polynomial: invalid length")
```

Comment:
If this is a generic Polynomial API: how do represent the zero polynomial? If you do it as `[0]`, then degree doesn’t match length-1 anymore.

Reply:
this implementation does not consider the representation of the zero polynomial.
It can be done by adding some checks here and there, but it's not needed for the use case. We might added if it's really needed.

Reply:
on a second thought, I added support to it, it wasn't so difficult after all.